sources/openmetrics/openmetrics.go

```diff
@@ -2,35 +2,156 @@ package openmetrics
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"io"
 	"net/http"
+	"net/url"
 	"regexp"
+	"sort"
+	"time"
 
 	dto "github.com/prometheus/client_model/go"
 	"github.com/prometheus/common/expfmt"
 	"github.com/sirupsen/logrus"
+	"github.com/stripe/veneur/v14"
+	"github.com/stripe/veneur/v14/samplers"
+	"github.com/stripe/veneur/v14/sources"
+	"github.com/stripe/veneur/v14/util"
 )
 
-type PrometheusResults struct {
+type OpenMetricsSourceConfig struct {
+	HistogramBucketTag string        `yaml:"histogram_bucket_tag"`
+	Host               util.Url      `yaml:"host"`
+	IgnoredMetrics     util.Regexp   `yaml:"ignored_metrics"`
+	Interval           time.Duration `yaml:"interval"`
+	SummaryQuantileTag string        `yaml:"summary_quantile_tag"`
+}
+
+type OpenMetricsSource struct {
+	cancelFunc         context.CancelFunc
+	context            context.Context
+	histogramBucketTag string
+	Host               *url.URL
```

Comment:
capital "H"?

Reply:
These are currently public due to use in `veneur-prometheus`. I added a TODO to make these private once `veneur-prometheus` is removed.